plugins/catalog-react/src/alpha/blueprints/EntityContextMenuItemBlueprint.tsx

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2025 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+import {
+  DialogApiDialog,
+  ExtensionBoundary,
+  coreExtensionData,
+  createExtensionBlueprint,
+  dialogApiRef,
+} from '@backstage/frontend-plugin-api';
+import MenuItem from '@material-ui/core/MenuItem';
+import ListItemIcon from '@material-ui/core/ListItemIcon';
+import ListItemText from '@material-ui/core/ListItemText';
+
+/** @alpha */
+export type FactoryLoaderParams = {
+  loader: () => Promise<JSX.Element>;
+};
+
+/** @alpha */
+export type FactoryHrefParams =
+  | {
+      title: string;
+      icon: JSX.Element;
+      useHref: () => string;
+    }
+  | {
+      title: string;
+      icon: JSX.Element;
+      href: string;
+    };
+
+/** @alpha */
+export type FactoryDialogParams = {
+  dialogLoader: () => Promise<
```

Comment:
Did you have a chance to discuss this parameter with @Rugvip? I'm curious if you guys considered an `onClick` parameter that would receive the apis?

Reply:
I've moved to the useOnClick approach you described and it seems to me we don't need to pass apis to it since hooks can be called directly from `useOnClick` as needed